content/blogs/YYYY-MM-DD-CI-CD-kestra-comparison.md

```diff
@@ -0,0 +1,217 @@
+
+In particular, CI/CD tools are designed to:
+
+- **Automate builds**: Compile source code into executable programs, saving developers time and minimizing manual errors.
+- **Run tests**: Execute automated tests to verify that code changes work as intended and don't introduce new bugs, providing early feedback and maintaining code quality.
+- **Deploy applications**: Automatically release new code to production or staging environments, allowing teams to ship features faster and with fewer manual interventions.
+- **Provide feedback**: Alert developers about build statuses and test results, helping them identify and address issues as soon as possible.
+
+These tools are fundamental for maintaining high-quality code and rapid release cycles in today's fast-paced development environments. Without them, teams would struggle to keep up with the demand for frequent updates, making it much harder to maintain a competitive edge.
+
+## Most Used CI/CD Tools
+There are a lot of CI/CD tools out there, so let's take a closer look at some of the most popular ones. For each, we'll provide an overview, highlight a unique feature, and list pros and cons.
+
+### GitHub Actions
+[GitHub Actions](https://github.com/features/actions) is GitHub's integrated CI/CD solution, allowing you to automate your workflows directly from your GitHub repository. It enables you to build, test, and deploy your code right alongside your pull requests and issues, providing a seamless experience for developers.
+
+Also, by integrating tightly with GitHub, it reduces context switching, allowing teams to manage their entire CI/CD pipelines within a single interface; this makes GitHub Actions particularly attractive for teams that want to quickly implement automation without having to leave their existing GitHub environment.
+
+![A use of GitHub Actions by Federico Trotta](/blogs/YYYY-MM-DD-CI-CD-kestra-comparison/github_actions.png)
+
+**Unique feature: deep integration with the GitHub ecosystem**
+GitHub Actions offers seamless integration with the entire GitHub platform, letting you trigger workflows based on any GitHub event—like pull requests, issues, or comments. This makes it an excellent choice for projects already using GitHub, as it provides a highly streamlined experience.
+
+**Pros**:
+- **Ease of use**:Simple setup within your GitHub repository, which means minimal configuration.
+- **Marketplace for actions**: Access to a vast library of pre-built actions, which can save you a lot of time..
+- **Cost-effective**: Generous free tier for public repositories, making it accessible for open-source projects.
+
+**Cons**:
+- **Limited to gitHub**: Ideal for projects fully hosted on GitHub, but the limited integration with other platforms can make it challenging for teams managing repositories across multiple environments.
+- **Debugging challenges**: Requires commits to test workflows, which can clutter the commit history and make it harder to maintain a clean repository.

```

Comment:
Technically you can trigger them in the Actions page. It's more that to edit your workflow, you also have to commit to the repository. Annoying because now your commit history has a combination of your "business logic" (or codebase for this article) and "orchestration" (or CI/CD for this article) logic combined together